services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/resource_tracker_pricing_plans.py

```diff
@@ -1,60 +1,83 @@
 from typing import Annotated
 
-from fastapi import Depends, Query
-from models_library.api_schemas_webserver.resource_usage import (
-    PricingDetailMinimalGet,
-    PricingPlanGet,
+from fastapi import Depends
+from models_library.api_schemas_resource_usage_tracker.pricing_plans import (
+    PricingUnitGet,
+    ServicePricingPlanGet,
 )
 from models_library.products import ProductName
+from models_library.resource_tracker import PricingPlanId, PricingUnitId
 from models_library.services import ServiceKey, ServiceVersion
 
 from ..api.dependencies import get_repository
 from ..modules.db.repositories.resource_tracker import ResourceTrackerRepository
 
 
-async def list_pricing_plans(
+async def get_service_default_pricing_plan(
     product_name: ProductName,
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
     resource_tracker_repo: Annotated[
         ResourceTrackerRepository, Depends(get_repository(ResourceTrackerRepository))
     ],
-    service_key: ServiceKey = Query(None),
-    service_version: ServiceVersion = Query(None),
-) -> list[PricingPlanGet]:
-    if service_key is None and service_version is None:
-        active_pricing_plans = (
-            await resource_tracker_repo.list_active_pricing_plans_by_product(
-                product_name
-            )
-        )
-    else:
-        active_pricing_plans = await resource_tracker_repo.list_active_pricing_plans_by_product_and_service(
-            product_name, service_key, service_version
-        )
+) -> ServicePricingPlanGet:
+    active_service_pricing_plans = await resource_tracker_repo.list_active_service_pricing_plans_by_product_and_service(
+        product_name, service_key, service_version
+    )
 
-    list_of_pricing_plan_get = []
-    for active_pricing_plan in active_pricing_plans:
-        list_of_pricing_detail_db = (
-            await resource_tracker_repo.list_pricing_details_by_pricing_plan(
-                active_pricing_plan.pricing_plan_id
-            )
+    default_pricing_plan = None
+    for active_service_pricing_plan in active_service_pricing_plans:
+        if active_service_pricing_plan.service_default_plan is True:
+            default_pricing_plan = active_service_pricing_plan
+            break
+
+    if default_pricing_plan is None:
+        raise ValueError("No default pricing plan for the specified service")
```

Comment:
Maybe use a custom exception derived from a base exception used in thisservice. Better yet if the base inherits from `PydanticErrorMixin`

Reply:
Thanks, done :+1: